clients/fides-js/src/lib/preferences.ts

```diff
@@ -276,44 +289,73 @@ const validateConsent = (fides: FidesGlobal, consent: NoticeConsent) => {
  * If both are provided, fidesString takes priority.
  * Can be used as a convenience method to update consent preferences using the FidesGlobal object.
  */
+export interface UpdateConsentOptions {
+  noticeConsent?: NoticeConsent;
+  fidesString?: string;
+  validation?: UpdateConsentValidation;
+  consentMethod?: ConsentMethod;
+  eventExtraDetails?: FidesEventExtraDetails;
+  tcf?: TcfSavePreferences;
+  updateCookie?: (oldCookie: FidesCookie) => Promise<FidesCookie>;
+}
 export const updateConsent = async (
-  fides: FidesGlobal,
-  options: {
-    consent?: NoticeConsent;
-    fidesString?: string;
-    validation?: UpdateConsentValidation;
-  },
-  consentMethod: ConsentMethod = ConsentMethod.SCRIPT,
-  eventExtraDetails: FidesEventExtraDetails = {
-    trigger: {
-      origin: FidesEventOrigin.EXTERNAL,
-    },
-  },
+  context: Pick<FidesGlobal, "experience" | "cookie" | "config" | "locale">,
+  consentOptions: UpdateConsentOptions,
+  servedNoticeHistoryId?: string,
 ): Promise<void> => {
-  // If neither consent nor fidesString is provided, raise an error
-  if (!options?.consent && !options?.fidesString) {
-    throw new Error("Either consent or fidesString must be provided");
-  }
-  if (!fides.experience) {
+  const { experience, cookie, config, locale } = context;
+  if (!experience) {
     throw new Error("Experience must be initialized before updating consent");
   }
-  if (!fides.cookie) {
+  if (!config) {
+    throw new Error("Config is not initialized");
+  }
+  if (!cookie) {
     throw new Error("Cookie is not initialized");
   }
-  const {
-    consent,
-    fidesString,
-    validation = UpdateConsentValidation.THROW,
-  } = options;
-
-  if (!Object.values(UpdateConsentValidation).includes(validation)) {
+  // If neither consent nor fidesString is provided, raise an error
+  if (!consentOptions?.noticeConsent && !consentOptions?.fidesString) {
+    throw new Error("Either consent object or fidesString must be provided");
+  }
```

Comment:
Does this work for TCF, when there are no custom notices? I'm not seeing a `fidesString` passed in from TCF but maybe I'm mistaken

Reply:
Good catch! It does work right now because we're at least passing in an empty object as `noticeConsent` from the TcfOverlay. However, that's putting the burden on the caller and it's best to also check for `tcf` here. I'll update that. Note: this was originally to alert customers that their configuration was wrong when using `window.Fides.updateConsent()`, which doesn't currently support TCF. For that reason I'm not going to update the message to match. I'll make note of that in the code as well.